addons/project/models/project_project.py

```diff
@@ -51,24 +51,17 @@ def _search_is_favorite(self, operator, value):
             raise NotImplementedError(_('Operation not supported'))
         return [('favorite_user_ids', 'in' if (operator == '=') == value else 'not in', self.env.uid)]
 
+    @api.depends('favorite_user_ids')
     def _compute_is_favorite(self):
         for project in self:
             project.is_favorite = self.env.user in project.favorite_user_ids
 
-    def _inverse_is_favorite(self):
-        favorite_projects = not_fav_projects = self.env['project.project'].sudo()
-        for project in self:
-            if self.env.user in project.favorite_user_ids:
-                favorite_projects |= project
-            else:
-                not_fav_projects |= project
-
-        # Project User has no write access for project.
-        not_fav_projects.write({'favorite_user_ids': [(4, self.env.uid)]})
-        favorite_projects.write({'favorite_user_ids': [(3, self.env.uid)]})
-
-    def _get_default_favorite_user_ids(self):
-        return [(6, 0, [self.env.uid])]
+    def _set_favorite_user_ids(self):
+        self_sudo = self.sudo()
+        if self.env.user in self.favorite_user_ids:
+            self_sudo.write({'favorite_user_ids': [(3, self.env.uid)]})
+        else:
+            self_sudo.write({'favorite_user_ids': [(4, self.env.uid)]})
```

Comment:
Rather than a simple toggle, which is kind of wonky, I think we could add a second argument to this method:
```suggestion
    def _set_favorite_user_ids(self, is_favorite):
        self_sudo = self.sudo() # To allow project users to set projects as favorite
        if is_favorite:
            self_sudo.write({'favorite_user_ids': [Command.link(self.env.uid)]})
        else:
            self_sudo.write({'favorite_user_ids': [Command.unlink(self.env.uid)]})
```
While I was at it, I added some very short documentation to justify the use of sudo and replaced the magic numbers with the `Command` class (which needs to be imported).
Also, to check, but I don't think we need to use `write` anymore